packages/design-system/src/DatePicker/DatePicker.tsx

```diff
@@ -200,27 +202,43 @@ export function createDefaultShortcuts(
 
   return [
     ...singleDayShortcuts,
-    createShortcut(useSingleDateShortcuts ? "1 week ago" : "Past week", [
-      oneWeekAgo,
-      today,
-    ]),
-    createShortcut(useSingleDateShortcuts ? "1 month ago" : "Past month", [
-      oneMonthAgo,
-      today,
-    ]),
-    createShortcut(useSingleDateShortcuts ? "3 months ago" : "Past 3 months", [
-      threeMonthsAgo,
-      today,
-    ]),
+    ...(excludeShortcuts.includes("past_week")
+      ? []
+      : [
+          createShortcut(useSingleDateShortcuts ? "1 week ago" : "Past week", [
+            oneWeekAgo,
+            today,
+          ]),
+        ]),
+    ...(excludeShortcuts.includes("past_month")
+      ? []
+      : [
+          createShortcut(
+            useSingleDateShortcuts ? "1 month ago" : "Past month",
+            [oneMonthAgo, today],
+          ),
+        ]),
+    ...(excludeShortcuts.includes("past_3_months")
+      ? []
+      : [
+          createShortcut(
+            useSingleDateShortcuts ? "3 months ago" : "Past 3 months",
+            [threeMonthsAgo, today],
+          ),
+        ]),
     // Don't include a couple of these for the single date shortcut
-    ...(useSingleDateShortcuts
+    ...(useSingleDateShortcuts || excludeShortcuts.includes("past_6_months")
       ? []
       : [createShortcut("Past 6 months", [sixMonthsAgo, today])]),
-    createShortcut(useSingleDateShortcuts ? "1 year ago" : "Past year", [
-      oneYearAgo,
-      today,
-    ]),
-    ...(useSingleDateShortcuts
+    ...(useSingleDateShortcuts || excludeShortcuts.includes("past_year")
```

Comment:
@tanvibhakta may be after sometime we should change this useSingleDateShortcuts flag.